### PR TITLE
feat(pandasai_server): pandasai server code commit

### PR DESCRIPTION
**Pull request by ArslanSaleem**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,7 +23,7 @@ coverage:  ## run unit tests and generate coverage report
 # SPELLCHECK AND FORMATTING
 ###########################
 
-IGNORE_FORMATS ?= "*.csv,*.txt,*.lock,*.log"
+IGNORE_FORMATS ?= "*.csv,*.txt,*.lock,*.log,server/app/models/user.py"
```
Comment (gventuri) [inline Makefile:26]: We shouldn't ignore specific files imho @ArslanSaleem

File: server/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+## PandasAI Server
```
Comment (gventuri) [inline server/README.md:1]: Would be interesting a quick overview on how to run the server and a brief overall description

File: server/app/controllers/space.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+from typing import List
```
Comment (gventuri) [inline server/app/controllers/space.py:1]: Let's rename it to "Workspace" :)

File: server/app/controllers/user.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        if not users:
+            await self.user_repository.create_and_init_dummy_user()
+
+    async def get_by_email(self, email: str) -> User:
+        return await self.user_repository.get_by_email(email)
+
+    async def me(self):
+        users = await self.get_all(limit=1)
+        if not users:
+            raise NotFoundException("No User found restart server and try again!")
```
Comment (gventuri) [inline server/app/controllers/user.py:30]: "No user found. Please restart the server and try again"

File: server/app/controllers/chat.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+        agent = Agent(connectors, config=config)
+
+        response = agent.chat(chat_request.query)
+
+        if os.path.exists(path_plot_directory):
+            shutil.rmtree(path_plot_directory)
+
+        if isinstance(response, str) and (
+            response.startswith("Unfortunately, I was not able to")
```
Comment (gventuri) [inline server/app/controllers/chat.py:64]: The message seems incomplete

File: server/app/controllers/auth.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        super().__init__(model=User, repository=user_repository)
+        self.user_repository = user_repository
+
+    @Transactional(propagation=Propagation.REQUIRED)
+    async def register(self, email: EmailStr, password: str, username: str) -> User:
+        # Check if user exists with email
+        user = await self.user_repository.get_by_email(email)
+
+        if user:
+            raise BadRequestException("User already exists with this email")
```
Comment (gventuri) [inline server/app/controllers/auth.py:23]: A user with this email already exists

File: server/app/controllers/auth.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+        # Check if user exists with username
+        user = await self.user_repository.get_by_username(username)
+
+        if user:
+            raise BadRequestException("User already exists with this username")
```
Comment (gventuri) [inline server/app/controllers/auth.py:29]: A user with this username already exists

File: server/app/controllers/chat.py (hunk 5 above)
Comment (ArslanSaleem) [reply server/app/controllers/chat.py:90]: No its error from library side when there is an error we return fallback message

Comment (gventuri) [reply server/app/controllers/chat.py:106]: I misunderstood, thanks a lot for clarifying

**Pull request merged**